lib/elastic_apm/config.rb

```diff
@@ -95,7 +95,7 @@ def initialize(options = {})
 
       yield self if block_given?
 
-      @logger = custom_logger || build_logger
+      @logger ||= custom_logger || build_logger
 
```

Comment:
Is it still possible for someone to set an agent custom logger when using Rails?
If someone has a Rails app but also a `logger` option in the config, the `@logger` would be set to the Rails logger by the time this line is reached. If I'm not mistaken, the `custom_logger` wouldn't be taken into account.

Reply:
There's a test: https://github.com/elastic/apm-agent-ruby/blob/master/spec/elastic_apm/config_spec.rb#L180-L186

All the default tries at getting loggers use `||=` so if a user sets one, nothing will be overridden.

Reply:
I wrote a test in the `rails_spec` that shows what I mean:
```ruby
  RSpec.describe 'custom agent logger', :mock_intake do
    before do
      class RailsApp < Rails::Application
        config.elastic_apm.logger = Logger.new(nil)
        config.logger = Logger.new(nil)
      end

      MockIntake.instance.stub!
      RailsApp.initialize!
    end

    it 'sets the custom logger' do
      expect(Rails.logger).not_to be(ElasticAPM.agent.config.logger)
    end
  end
```

This fails, as the logger on the agent config is set to the Rails logger in [this line](https://github.com/elastic/apm-agent-ruby/blob/master/lib/elastic_apm/config.rb#L226). Is that the expected behavior -- that you wouldn't be able to have a custom logger for the agent if using Rails?

Reply:
Added your test and addressed it 👍

Reply:
Looks good!